### PR TITLE
Introduce createStore function

### PR DESCRIPTION
**Pull request by DmitryEfimenko**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/elf/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Store can be created in two steps:
```ts
const { state, config } = createState(withProps<AuthProps>({ user: null }));
const authStore = new Store({ state, name: 'auth', config });
```
This behavior is still there.

Issue Number: N/A

## What is the new behavior?

* Store can be created in one step:
  ```ts
  const store = createStore({ name: 'auth' }, withProps<AuthProps>({ user: null }));
  ```

* The Elf CLI is updated to generate files using this approach.
* READMEs and docs, and demos are updated to use this approach

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

## Other information
* Also installed and added to dev dependencies 3 packages that were for some reason missing. Without them I wasn't able to run tests.
* Not sure why PR shows 7 commits. There should only be one. I'm not a git expert. I double-checked and made sure that only the relevant changes are included.

### CODE REVIEW

File: packages/store/src/lib/create-store.spec.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { withEntities } from '@ngneat/elf-entities';
+import { Todo } from '@ngneat/elf-mocks';
+import { createStore } from './create-store';
+import { withProps } from './props.state';
+
+describe('createStore', () => {
+  it('should create store', () => {
+    const store = createStore(
```
Comment (NetanelBasal) [inline packages/store/src/lib/create-store.spec.ts:8]: Add specs for the return types of state and config, please.

File: package.json [hunk 2]
```diff
@@ -68,6 +68,7 @@
     "@nrwl/tao": "13.8.0",
     "@nrwl/web": "13.8.0",
     "@nrwl/workspace": "13.8.0",
+    "@oclif/command": "^1.8.16",
```
Comment (NetanelBasal) [inline package.json:71]: revert deps

Comment (DmitryEfimenko) [reply package.json:71]: I can certainly do it, but I wasn't able to run tests locally without these installed. Are you sure? If so, why?

Comment (NetanelBasal) [reply package.json:71]: You need to run `npm i` inside packages/cli. We should use npm workspaces and it'll be automatically.

File: packages/cli/src/builders/repo-builder.ts [hunk 3]
```diff
@@ -70,10 +70,14 @@ export function createRepo(options: Options) {
     }
   }
 
-  const state = factory.createCallExpression(
-    factory.createIdentifier('createState'),
+  const storeOpts = factory.createIdentifier(
```
Comment (NetanelBasal) [inline packages/cli/src/builders/repo-builder.ts:73]: @EricPoul note that you'll have conflicts

**Review by NetanelBasal — APPROVED** [review]

**Comment by NetanelBasal** [comment]
@DmitryEfimenko please squash the commits

**Comment by DmitryEfimenko** [comment]
I'm not a git expert and having troubles with this... running command `git rebase -i HEAD~2`, picking first commit and squashing second commit. But that results in branch diverging. After doing `git pull`, everything goes back...

**Comment by DmitryEfimenko** [comment]
I think I figured out the squashing...
Although I'm not sure if you want to me squash the "merge master into dev" commit as well. Let me know

**Comment by NetanelBasal** [comment]
Something is wrong with your git history:

image

Maybe you can create a new clean branch from master and cherry-pick from yours.

**Comment by EricPoul** [comment]
@DmitryEfimenko please check the last version of cli, there can be conflicts.

**Comment by DmitryEfimenko** [comment]
I'll close this PR and re-create it fresh

**Pull request closed without merging**